cli/run_util.go

```diff
@@ -426,18 +430,45 @@ func (r WorkflowRunner) executeStep(
 	logger := log.NewLogger(opts)
 	stdout := logwriter.NewLogWriter(logger)
 
-	name := cmdArgs[0]
+	var name string
 	var args []string
-	if len(cmdArgs) > 1 {
-		args = cmdArgs[1:]
+	var envs []string
+
+	if workflow.Container.Image != "" {
+		envs, err = envman.ReadAndEvaluateEnvs(configs.InputEnvstorePath, &docker.DockerEnvironmentSource{
+			Logger: logger,
+		})
+		if err != nil {
+			return 1, fmt.Errorf("failed to read command environment: %w", err)
+		}
+
+		name = "docker"
+		container := r.dockerManager.GetWorkflowContainer(workflowID)
+		if container == nil {
+			return 1, fmt.Errorf("Docker container does not exist")
```

Comment:
I have a feeling this error is going to be printed often and we should add more details and troubleshooting ideas if possible.

Reply:
On the contrary this will only be shown if there was some failure in starting the container which happens before the steps start to execute. And in that case we have verbose logging in place showing the cause of that problem. It is not necessary to add any more here.